nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,336 @@
+          GrantType = "client_credentials";
+        };
+        ExtraConfig = cfg.idpManagerExtraConfig;
+      };
+      DeviceAuthorizationFlow = {
+        Provider = NETBIRD_AUTH_DEVICE_AUTH_PROVIDER;
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_DEVICE_AUTH_AUDIENCE;
+          Domain = NETBIRD_AUTH0_DOMAIN;
+          ClientID = NETBIRD_AUTH_DEVICE_AUTH_CLIENT_ID;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          DeviceAuthEndpoint = NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_DEVICE_AUTH_SCOPE;
+          UseIDToken = NETBIRD_AUTH_DEVICE_AUTH_USE_ID_TOKEN;
+        };
+      };
+      PKCEAuthorizationFlow = {
+        ProviderConfig = {
+          Audience = NETBIRD_AUTH_AUDIENCE;
+          ClientID = NETBIRD_AUTH_CLIENT_ID;
+          ClientSecret = NETBIRD_AUTH_CLIENT_SECRET;
+          AuthorizationEndpoint = NETBIRD_AUTH_PKCE_AUTHORIZATION_ENDPOINT;
+          TokenEndpoint = NETBIRD_AUTH_TOKEN_ENDPOINT;
+          Scope = builtins.concatStringsSep " " NETBIRD_AUTH_SUPPORTED_SCOPES;
+          RedirectURLs = NETBIRD_AUTH_PKCE_REDIRECT_URLS;
+          UseIDToken = NETBIRD_AUTH_PKCE_USE_ID_TOKEN;
+        };
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = [ 80 443 33073 10000 ];
+
+    services.nginx.virtualHosts = {
+      ${cfg.settings.NETBIRD_DOMAIN} = {
+        forceSSL = true;
+        enableACME = true;
+
+        locations."/" = {
+          root = "${stateDir}/web-ui/";
+          tryFiles = "$uri /index.html";
+        };
+
+        locations."/signalexchange.SignalExchange/".extraConfig = ''
+          grpc_pass grpc://localhost:${builtins.toString cfg.ports.signal};
+          # grpc_ssl_verify off;
+          grpc_read_timeout 1d;
+          grpc_send_timeout 1d;
+          grpc_socket_keepalive on;
+        '';
+
+        locations."/api".proxyPass = "http://localhost:${builtins.toString cfg.ports.management}";
+
+        locations."/management.ManagementService/".extraConfig = ''
+          grpc_pass grpc://localhost:${builtins.toString cfg.ports.management};
+          # grpc_ssl_verify off;
+          grpc_read_timeout 1d;
+          grpc_send_timeout 1d;
+          grpc_socket_keepalive on;
+        '';
+      };
+    };
+
+    systemd.services.netbird-setup = {
+      wantedBy = [ "netbird-management.service" "netbird-signal.service" "multi-user.target" ];
+      serviceConfig = {
+        Type = "oneshot";
+        RuntimeDirectory = "netbird-mgmt";
+        StateDirectory = "netbird-mgmt";
+        WorkingDirectory = stateDir;
+        EnvironmentFile = [ settingsFile ];
+      };
+      unitConfig = {
+        StartLimitInterval = 5;
+        StartLimitBurst = 10;
+      };
+
+      path = with pkgs; [ coreutils findutils gettext gnused ];
+
+      script = ''
+        cp ${managementFile} ${stateDir}/config.json
+        cat ${stateDir}/config.json
+
+        mkdir -p ${stateDir}/data
```

Comment:
I suggest to do `StateDirectory = [ "netbird-mgmt" "netbird-mgmt/data" ]` which should work I believe